src/templates/dandelion/config/screens/TokenRequestScreen.js

```diff
@@ -0,0 +1,243 @@
+import React, { useCallback, useState, useRef } from 'react'
+import PropTypes from 'prop-types'
+import { addressesEqual, Field, GU, Help, Info, isAddress } from '@aragon/ui'
+import {
+  Header,
+  Navigation,
+  ScreenPropsType,
+  KnownAppBadge,
+} from '../../../kit'
+import MultiTokenSelector from '../../components/TokenSelector/MultiTokenSelector'
+import { getDefaultAcceptedTokens } from '../helpers/tokens'
+import { shortenAddress } from '../../../../web3-utils'
+
+function useFieldsLayout() {
+  // In its own hook to be adapted for smaller views
+  return `margin-bottom: ${2 * GU}px`
+}
+
+function validationError(acceptedTokens) {
+  if (acceptedTokens.length === 0) {
+    return 'You need to select at least one accepted token.'
+  }
+
+  const notValidAddress = acceptedTokens.some(
+    ({ token }) => !isAddress(token.address)
+  )
+
+  if (notValidAddress) {
+    return 'One or more accepted tokens are not valid addresses.'
+  }
+
+  return null
+}
+
+const EMPTY_TOKEN = { token: { address: '' }, selectedIndex: -1 }
+
+function TokenRequestScreen({
+  appLabel,
+  dataKey,
+  screenProps: { back, data, next, screenIndex, screens },
+  title,
+}) {
+  const screenData = (dataKey ? data[dataKey] : data) || {}
+
+  const [acceptedTokens, setAcceptedTokens] = useState(
+    screenData.acceptedTokens && screenData.acceptedTokens.length > 0
+      ? screenData.acceptedTokens
+      : [EMPTY_TOKEN]
+  )
+
+  const [formError, setFormError] = useState()
```

Comment:
Should this start off `null`? I think we forgot to do this in a few other places as well!